src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -875,18 +875,6 @@ HeapWord* ShenandoahHeap::allocate_memory(ShenandoahAllocRequest& req) {
       result = allocate_memory_under_lock(req, in_new_region);
     }
 
-    // Check that gc overhead is not exceeded.
-    //
-    // Shenandoah will grind along for quite a while allocating one
-    // object at a time using shared (non-tlab) allocations. This check
-    // is testing that the GC overhead limit has not been exceeded.
-    // This will notify the collector to start a cycle, but will raise
-    // an OOME to the mutator if the last Full GCs have not made progress.
-    if (result == nullptr && !req.is_lab_alloc() && get_gc_no_progress_count() > ShenandoahNoProgressThreshold) {
```

Comment:
Does this render `ShenandoahNoProgressThreshold` a dead option?

Reply:
Good point.  It is no longer relevant.

Reply:
I'll remove that option for coherency in this PR.   Still need to resolve the question of too-much-time for OOM, mentioned below.

Reply:
I've modified this code after further testing, and have resurrrected dependence on ShenandoahNoProgressThreshold, with a slightly different behavior than previously implemented.